### PR TITLE
Fix sigmoid deprecation

### PR DESCRIPTION
**Pull request by dme65**
- Importing sigmoid from ```torch``` instead of ```torch.nn.functional``` as the latter is deprecated
- Fixing a dtype error in ```PolynomialKernelGrad```

### CODE REVIEW

**Comment by jacobrgardner** [comment]
Rebased on to master (where the PolynomialKernelGrad stuff was already fixed).
